aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -680,16 +688,15 @@ Tensor dot_cuda(const Tensor& self, const Tensor& other) {
     incy = 1;
   }
 
-if (self._is_zerotensor() || other._is_zerotensor()) {
-  return at::_efficientzerotensor({}, self.options());
-}
+  if (self._is_zerotensor() || other._is_zerotensor()) {
+    result.fill_(0);
+    return;
```

Comment:
Same comment as before: how does this change the performance of this kernel?

Reply:
Same as above: could you please point us at which workflow we need to trigger to get those numbers?

Reply:
https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#cuda-development-tips

Reply:
The behavior is changed - before, this path would use ZeroTensor allocator: https://github.com/pytorch/pytorch/blob/ff616c26fb66b26966b6f6f50aa85bf615400816/aten/src/ATen/native/TensorFactories.h#L119-L120, which seems to not fill result with 0.